<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
<title>polyfreqs_pps. polyfreqs 1.0.0</title>
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<meta name="author" content="">

<link href="css/bootstrap.css" rel="stylesheet">
<link href="css/bootstrap-responsive.css" rel="stylesheet">
<link href="css/highlight.css" rel="stylesheet">
<link href="css/staticdocs.css" rel="stylesheet">

<!--[if lt IE 9]>
  <script src="http://html5shim.googlecode.com/svn/trunk/html5.js"></script>
<![endif]-->

<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    tex2jax: {
      inlineMath: [ ['$','$'], ["\\(","\\)"] ],
      processEscapes: true
    }
  });
</script>
<script type="text/javascript"
  src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML">
</script>
  </head>

  <body>
    <div class="navbar">
  <div class="navbar-inner">
    <div class="container">
      <a class="brand" href="#">polyfreqs 1.0.0</a>
      <div class="nav">
        <ul class="nav">
          <li><a href="index.html"><i class="icon-home icon-white"></i> Index</a></li>
        </ul>
      </div>
    </div>
  </div>
</div>

    <div class="container">
      <header>
        
      </header>
      
      <h1>Posterior predictive model checks for polyfreqs</h1>

<div class="row">
  <div class="span8">
    <h2>Usage</h2>
    <pre><div>polyfreqs_pps(p_post, tM, rM, ploidy, error)</div></pre>
    
    <h2>Arguments</h2>
    <dl>
      <dt>p_post</dt>
      <dd>A matrix containing the posterior samples from a <code><a href='polyfreqs.html'>polyfreqs</a></code> run.</dd>
      <dt>tM</dt>
      <dd>Total reads matrix: matrix containing the total number of reads mapping to each locus for each individual.</dd>
      <dt>rM</dt>
      <dd>Reference reads marix: matrix containing the number of reference reads mapping to each locus for each individual.</dd>
      <dt>ploidy</dt>
      <dd>Ploidy level of individuals in the population.</dd>
      <dt>error</dt>
      <dd>The level of sequencing error. A fixed constant.</dd>
    </dl>
    
    <div class="Value">
      <h2>Value</h2>

      <p><dl>
A list with two items:
<dl>
 <dt>ratio_diff</dt><dd>The posterior predictive samples of the difference between the simulated read ratios and the observed read ratio summed across individuals at each locus.</dd></p>
  
      <p><dt>locus_fit</dt><dd>A logical vector indicating whether or not each locus passed or failed the posterior predictive model check.</dd></p>
  
      <p></dl></p>
  
      <p></dl></p>
  
    </div>

    <div class="Description">
      <h2>Description</h2>

      <p>Uses the posterior distribution of allele frequences from a <code><a href='polyfreqs.html'>polyfreqs</a></code> run to test model fit using the posterior predictive model checking procedure described in Blischak <em>et al</em>.</p>
  
    </div>

    <div class="Details">
      <h2>Details</h2>

      <p>The observed read count ratio (r/t) for each locus is summed across individuals and then compared to a distribution of read ratios simulated using the posterior allele frequencies by taking their difference.
The criterion for passing/failing the posterior predictive check is then made on a per locus basis based on whether or not the distribution of read ratio differences contains 0 in the 95</p>
  
    </div>

    <div class="References">
      <h2>References</h2>

      <p>Blischak PD, LS Kubatko and AD Wolfe. Accounting for genotype uncertainty in the estimation of allele frequencies in autopolyploids. <em>In revision</em>.</p>
  
    </div>
      </div>
  <div class="span4">
    <!-- <ul>
      <li>polyfreqs_pps</li>
    </ul>
    <ul>
      
    </ul> -->
      
        
  </div>
</div>
      
      <footer>
      <p class="pull-right"><a href="#">Back to top</a></p>
<p>Built by <a href="https://github.com/hadley/staticdocs">staticdocs</a>. Styled with <a href="http://twitter.github.com/bootstrap">bootstrap</a>.</p>
      </footer>
    </div>
  </body>
</html>