<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
<title>sim_reads. polyfreqs 1.0.0</title>
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<meta name="author" content="">

<link href="css/bootstrap.css" rel="stylesheet">
<link href="css/bootstrap-responsive.css" rel="stylesheet">
<link href="css/highlight.css" rel="stylesheet">
<link href="css/staticdocs.css" rel="stylesheet">

<!--[if lt IE 9]>
  <script src="http://html5shim.googlecode.com/svn/trunk/html5.js"></script>
<![endif]-->

<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    tex2jax: {
      inlineMath: [ ['$','$'], ["\\(","\\)"] ],
      processEscapes: true
    }
  });
</script>
<script type="text/javascript"
  src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML">
</script>
  </head>

  <body>
    <div class="navbar">
  <div class="navbar-inner">
    <div class="container">
      <a class="brand" href="#">polyfreqs 1.0.0</a>
      <div class="nav">
        <ul class="nav">
          <li><a href="index.html"><i class="icon-home icon-white"></i> Index</a></li>
        </ul>
      </div>
    </div>
  </div>
</div>

    <div class="container">
      <header>
        
      </header>
      
      <h1>Simulation of sequencing read counts and genotypes</h1>

<div class="row">
  <div class="span8">
    <h2>Usage</h2>
    <pre><div>sim_reads(pVec, N_ind, coverage, ploidy, error)</div></pre>
    
    <h2>Arguments</h2>
    <dl>
      <dt>pVec</dt>
      <dd>A vector of allele frequencies strung together using the concatenate function.</dd>
      <dt>N_ind</dt>
      <dd>The number of individuals to simulate.</dd>
      <dt>coverage</dt>
      <dd>The average number of sequences simulated per individual per locus (Poisson distributed).</dd>
      <dt>ploidy</dt>
      <dd>The ploidy level of individuals in the population.</dd>
      <dt>error</dt>
      <dd>The level of sequencing error. A fixed constant.</dd>
    </dl>
    
    <div class="Value">
      <h2>Value</h2>

      <p><dl>
A list of 3 matrices:
<dl>
 <dt><code>genos</code></dt><dd>A matrix of the simulated genotypes.</dd></p>
  
      <p><dt><code>tot_read_mat</code></dt><dd>A matrix of the simulated number of total reads.</dd></p>
  
      <p><dt><code>ref_read_mat</code></dt><dd>A matrix of the simulated number of reference reads.</dd></p>
  
      <p></dl></p>
  
      <p></dl></p>
  
    </div>

    <div class="Description">
      <h2>Description</h2>

      <p>Simulates genotypes and read counts under the model of Blischak <em>et al</em>.</p>
  
    </div>

    <div class="Details">
      <h2>Details</h2>

      <p>Total reads are simulated using a Poisson distribution with mean equal to the <code>coverage</code> set by the user.
Next, genotypes are simulated for the specified number of individuals using the vector of allele frequencies provided to the function.
The number of loci simulated is equal to the number of elements supplied by the vector of allele frequencies.
The number of reference reads is then simulated using Eq. 1 from Blischak <em>et al</em>. using the total reads, genotypes and sequencing error.</p>
  
    </div>

    <div class="References">
      <h2>References</h2>

      <p>Blischak PD, Kubatko LS, Wolfe AD. 2015. Accounting for genotype uncertainty in the estimation of allele frequencies in autopolyploids. <em>In review</em>. bioRxiv, <strong>doi</strong>:####.</p>
  
    </div>
      </div>
  <div class="span4">
    <!-- <ul>
      <li>sim_reads</li>
    </ul>
    <ul>
      
    </ul> -->
      
        
  </div>
</div>
      
      <footer>
      <p class="pull-right"><a href="#">Back to top</a></p>
<p>Built by <a href="https://github.com/hadley/staticdocs">staticdocs</a>. Styled with <a href="http://twitter.github.com/bootstrap">bootstrap</a>.</p>
      </footer>
    </div>
  </body>
</html>